### DISCUSSION TITLE
How to handle simple objects?

### DISCUSSION THREAD

**Discussion by lewysigns** [anvilistas/tabulator#55] [General]
Would a custom formatter be needed to access list of links?

now sure how this would work in practice. Here is my go at it:

table  1 scheme:

column 1 : [ link to table 2 , link to table 2]
column 2 : index

table 2 scheme:

name: "NAME"

```
columns =  [
    {'title':"Test", 'field':"column 1",'formatter':self.cust_formatter}
]
def cust_formatter(cell,**params):
    # somehow need column 2 value
     values =     cell.get_value()
   return values[column 2]['name']

```

**Comment by s-cork** [comment]
I'm assuming you're using the `app_tables` option?

Is column1 a multi linked table row?

If it's a single link you should be able to do something like:
```python
columns =  [
    {'title':"Link", 'field':"my_link.name"}
]
```
If it's multilink you could do stick a `my_links.0.name` but that's not necessarily the best option (i'm not sure it'll even work)

If you're using the `app_tables` option then you can also provide a custom getter

```python
def custom_getter(row, field):
    if field == 'my_link':
        return row['my_link']['name']
    return row[field]

...

    self.tabulator.options['getter'] = custom_getter
```

And yes a custom formatter will also work.

If that doesn't help, some more details and a clone link would be helpful

**Comment by lewysigns** [comment]
I was able to successfully index my multi-link column with a custom formatter. I tried to use the app_tables option, but in the custom formatter I received the following error:

![image](https://user-images.githubusercontent.com/66562010/170572567-128b8ec6-3ac2-418d-9666-88c57a31ede3.png)


I believe this was due to trying to access a  ```<LiveObject: anvil.tables.Row>``` in the custom formatter. So instead I (very poorly) converted the data to a dictionary in my server method ```get_data```. Another thing I am worried about using app_tables option is permission/filtering the data before the client receives the data (in production).

https://anvil.works/build#clone:TX5IJADW5BLXJTRB=IGU6N4GJMVJGZXZHMS4NRCBH|TGQCF3WT6FVL2EM2=

**Reply by lewysigns** [reply]
The custom formatter causes an ```[An internal error has occurred]``` to raise when trying to sort the column, so I'll call it a half solution.

**Reply by s-cork** [reply]
thanks for the clone

I wasn't necessarily suggesting to use the app_tables option, I just wasn't sure if you were or not.

In terms of the internal error,
since you provided a field for that column
Tabulator's sorter is trying to sort by the `links` field, but, the data there is a list, and Tabulator doesn't know how to sort lists.
If you want it to be sortable you'll have to provide a custom sorter.
If you don't want it to be sortable you can remove sorting for that column in the column definition.

**Reply by s-cork** [reply]
Also hint for internal errors, you can open up the browser console to see the error.
You can also run a published version of the app which will correctly report the error.

**Reply by lewysigns** [reply]
Thank you very much for that!
